src/main.c

```diff
@@ -18,7 +24,44 @@ static void repl(VM* vm) {
       break;
     }
 
-    interpret(vm, line);
+    int lBraceCount = 0;
+    int rBraceCount = 0;
+
+    for (int i = 0; line[i]; i++) lBraceCount += (line[i] == '{');
+    for (int i = 0; line[i]; i++) rBraceCount += (line[i] == '}');
+
+    if (lBraceCount > rBraceCount) {
+      char code[1024];
+
+      memcpy(code, line, sizeof(line));
+
+      for (;;) {
+        printf(".. ");
+
+        if (!fgets(line, sizeof(line), stdin)) {
+          printf("\n");
+          break;
+        }
+
+        int lBraceCount = 0;
+        int rBraceCount = 0;
+
+        for (int i = 0; line[i]; i++) lBraceCount += (line[i] == '{');
+        for (int i = 0; line[i]; i++) rBraceCount += (line[i] == '}');
```

Comment:
Why are you walking line twice? This can happen within a single loop

Reply:
Fixed!